lib/WP_Auth0_LoginManager.php

```diff
@@ -163,136 +170,124 @@ public function init_auth0() {
       } else {
         $this->redirect_login();
       }
-    } catch (WP_Auth0_LoginFlowValidationException $e) {
+    } catch ( WP_Auth0_LoginFlowValidationException $e ) {
 
+      // Errors during the OAuth login flow
       $this->die_on_login( $e->getMessage(), $e->getCode() );
 
-    } catch (WP_Auth0_BeforeLoginException $e) {
+    } catch ( WP_Auth0_BeforeLoginException $e ) {
 
+      // Errors during the WordPress login flow
       $this->die_on_login( $e->getMessage(), $e->getCode(), FALSE );
-
-    } catch (Exception $e) {
-
     }
-
   }
 
+  /**
+   * Main login flow, Authorization Code Grant
+   *
+   * @throws WP_Auth0_BeforeLoginException
+   * @throws WP_Auth0_LoginFlowValidationException
+   *
+   * @see https://auth0.com/docs/api-auth/tutorials/authorization-code-grant
+   */
   public function redirect_login() {
-    global $wp_query;
-
-    if ( $this->query_vars( 'auth0' ) === null ) {
-      return;
-    }
-
-    if ( $this->query_vars( 'error_description' ) !== null && $this->query_vars( 'error_description' ) !== '' ) {
-      throw new WP_Auth0_LoginFlowValidationException( sanitize_text_field( $this->query_vars( 'error_description' ) ) );
-    }
-
-    if ( $this->query_vars( 'error' ) !== null && trim( $this->query_vars( 'error' ) ) !== '' ) {
-      throw new WP_Auth0_LoginFlowValidationException( sanitize_text_field( $this->query_vars( 'error' ) ) );
-    }
-
-    $code = $this->query_vars( 'code' );
-    $state_decoded = $this->get_state();
 
     $domain = $this->a0_options->get( 'domain' );
 
     $client_id = $this->a0_options->get( 'client_id' );
     $client_secret = $this->a0_options->get( 'client_secret' );
 
-    if ( empty( $client_id ) ) {
-      throw new WP_Auth0_LoginFlowValidationException( __( 'Error: Your Auth0 Client ID has not been entered in the Auth0 SSO plugin settings.', 'wp-auth0' ) );
-    }
-    if ( empty( $client_secret ) ) {
-      throw new WP_Auth0_LoginFlowValidationException( __( 'Error: Your Auth0 Client Secret has not been entered in the Auth0 SSO plugin settings.', 'wp-auth0' ) );
-    }
-    if ( empty( $domain ) ) {
-      throw new WP_Auth0_LoginFlowValidationException( __( 'Error: No Domain defined in Wordpress Administration!', 'wp-auth0' ) );
-    }
-
-    $response = WP_Auth0_Api_Client::get_token( $domain, $client_id, $client_secret, 'authorization_code', array(
+    // Exchange authorization code for token
+    $exchange_resp = WP_Auth0_Api_Client::get_token( $domain, $client_id, $client_secret, 'authorization_code', array(
         'redirect_uri' => home_url(),
-        'code' => $code,
+        'code' => $this->query_vars( 'code' ),
       ) );
 
-    if ( $response instanceof WP_Error ) {
```

Comment:
Diff gets a little mangled from here down because of switching around how the errors are handled. This was removed in favor of checking a specific error code we're looking for, followed by a general check if we have the info we need (`access_token`)